### PR TITLE
Make email configs more ergonomical

### PR DESCRIPTION
**Pull request by aronerben**
This rework allows people to not have to cache their data they fetch from IO for configs

### CODE REVIEW

File: sihl-email/src/sihl_email.ml [hunk 1]
```diff
@@ -144,14 +136,15 @@ module MakeSmtp (Config : SmtpConfig) : Sihl.Contract.Email.Sig = struct
...
     in
-    let* sender = Config.sender ()
-    and* username = Config.username ()
-    and* password = Config.password ()
-    and* hostname = Config.hostname ()
-    and* port = Config.port ()
-    and* with_starttls = Config.start_tls ()
-    and* ca_path = Config.ca_path ()
-    and* ca_cert = Config.ca_cert () in
+    let* config = Config.config () in
```
Comment (jerben) [inline sihl-email/src/sihl_email.ml:139]: @aronerben If the module is already called `*Config` maybe we can call the function `fetch` or something like that? So the usage would be `Config.fetch()`.

**Review by jerben — CHANGES_REQUESTED** [review]
Is the configuration validated at service start-up time? If not, can you please add `let* _ = Config.fetch() in ...` to `start()`?

File: sihl-email/src/sihl_email.ml (hunk 1 above)
Comment (aronerben) [reply sihl-email/src/sihl_email.ml]: Yea you're right, adjusted!

**Comment by aronerben** [comment]
> Is the configuration validated at service start-up time? If not, can you please add `let* _ = Config.fetch() in ...` to `start()`?

Idk why, but I assumed that type checking would provide the same validation, but conformist does a lot of other stuff too, right? Anyway, I added it again for SendGrid and SMTP and only if in production and mails are not intercepted.

**Comment by jerben** [comment]
> Idk why, but I assumed that type checking would provide the same validation, but conformist does a lot of other stuff too, right? Anyway, I added it again for SendGrid and SMTP and only if in production and mails are not intercepted.

Yes conformist will tell you if the config is invalid but with the current implementation the `Config.fetch ()` is called the first time when an email is being sent right? Sihl services should not even start if the configuration they need is not present, so we need someone or something call either `fetch` or `require` as soon as the service starts. Kind of "fail as early as possible" like thinking.

**Review by jerben — APPROVED** [review]
Looks good, could you check my question? If the `require` is really needed it is ready for merge.

**Pull request merged**